dat.js

```diff
@@ -90,8 +94,9 @@ Dat.prototype.download = function (link, location, cb) {
       debug('done', entry.size)
     })
   })
-  metadata.on('error', cb)
-  metadata.on('end', function () {
```

Comment:
This was causing download to callback too early (and I think why share sometimes didn't work). The `end` event on metadata can fire before downloads are done.